src/foam/nanos/ruler/predicate/PropertyIsInstance.js

```diff
@@ -37,14 +37,21 @@ foam.CLASS({
     {
       name: 'f',
       javaCode: `
+      Class cls;
+      try {
+        cls = Class.forName(getOf());
```

Comment:
Try benchmarking this compared to the previous version. If there's no serious performance degradation then I'm good with this. Otherwise, cache the Class.